src/transform/ancestor.js

```diff
@@ -118,7 +118,7 @@ function canBeLifted(doc, range) {
   }
 }
 
-// :: (Node, Pos, ?Pos) → bool
+// :: (Node, Pos, ?Pos) → ?{found: any, range: any}
```

Comment:
By amending classes, you mean assign to their prototypes? Isn't that something that most JavaScript ends up doing? Though I guess doing outside of the class' own module might not be expected.

Rust has `-> !` for non-returning functions. That always worked well. But I guess Flow isn't that far along yet.

Does flow try to prove loop termination? I assumed that was out of scope.